ext/mysqli/mysqli_nonapi.c

```diff
@@ -744,13 +744,13 @@ static int mysqlnd_zval_array_to_mysqlnd_array(zval *in_array, MYSQLND ***out_ar
 			MYSQLI_RESOURCE *my_res;
 			mysqli_object *intern = Z_MYSQLI_P(elem);
 			if (!(my_res = (MYSQLI_RESOURCE *)intern->ptr)) {
-		  		php_error_docref(NULL, E_WARNING, "[%d] Couldn't fetch %s", i, ZSTR_VAL(intern->zo.ce->name));
-				continue;
+		  		zend_throw_error(NULL, "%s object is already closed", ZSTR_VAL(intern->zo.ce->name));
+				return -1;
```

Comment:
The function returns 0 by default, so I decided to use `-1` in case of an exceptional situation. I am not sure if it is ok like this or I should change them to 1/0 (using SUCCESS/FAILURE).

Reply:
SUCCESS/FAILURE would be appropriate here.